net/portmapper/portmapper.go

```diff
@@ -409,10 +414,18 @@ func (c *Client) createOrGetMapping(ctx context.Context) (external netaddr.IPPor
 		}
 	}
 
-	// And ask for a mapping.
-	pmpReqMapping := buildPMPRequestMappingPacket(localPort, prevPort, pmpMapLifetimeSec)
-	if _, err := uc.WriteTo(pmpReqMapping, pmpAddru); err != nil {
-		return netaddr.IPPort{}, err
+	if !haveRecentPMP && haveRecentPCP {
```

Comment:
(line 412 up there where stupid github won't let me comment)  .... you're still writing to PMP to ask for the external address in the PCP case.

how do we get the external IP with PCP?

Reply:
It's contained in the PCP packet, so we don't need to do anything special for it